src/lib.rs

```diff
@@ -30,17 +30,10 @@ trait Action {
     fn perform (&mut self, cx: Context<'_>) -> Result<Self::Output>;
```

Comment:
I was thinking something similar before, but decided against for the following reasons:

1. I couldn't find a good `std::io::ErrorKind` to fit and I'd prefer to just use `std::io::Result` for simplicity (also since these are IO actions, it makes sense to return only IO errors imo)
2. Outside of gross `unsafe` abuse, there should be no way to repeat an `Action` due to `IO`'s interface preventing it
3. End users can't actually call this function or provide its `Context` argument so there is no way for a user to mistakenly or willingly trigger a panic due to the requirement being violated